### ISSUE TITLE
Variable explorer sorting by Size

### ISSUE THREAD

**Issue by AbhishekVermaDH** [spyder-ide/spyder#14527]
```python
File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
```

In variable explorer when sorting by `Size` leads to this issue reporter. 

In RStudio size represents memory footprint of the object than dimensions as present in Spyder. So RStudio sorting make sense. For spyder showing `shape` of different objects such sorting is bit weird. Populating memory footprint in `Size` column will be better.

**Comment by ccordoba12** [comment]
Hey @AbhishekVermaDH, thanks for reporting. Please post a simple dataframe that generates this error so we can try to debug and fix it in our side.

Also, what's your Spyder version?

**Comment by Tissot11** [comment]
I have Spyder 4.2.1 installed on Big Sur. Everytime I click on an entry in the Variable Explorer pane, the whole Spyder window resize to a small size? What is the reason for this and how to fix it?

**Comment by AbhishekVermaDH** [comment]
Replication steps:
1. Use code
```
import pandas as pd
import numpy as np
import datetime
df = pd.DataFrame({
    'A' : ['spam', 'eggs', 'spam', 'eggs'] * 6,
    'B' : ['alpha', 'beta', 'gamma'] * 8,
    'C' : [np.random.choice(pd.date_range(datetime.datetime(2013,1,1),datetime.datetime(2013,1,3))) for i in range(24)],
    'D' : np.random.randn(24),
    'E' : np.random.randint(2,10,24),
    'F' : [np.random.choice(['rand_1', 'rand_2', 'rand_4', 'rand_6']) for i in range(24)],
})
lst = list(range(10))
```
2. So there will be 2 variables created in the `Variable explorer`. Try sorting them by `Size` by clicking on text.  This will lead to following error screen with stacktrace:
```
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 1708, in lessThan
    return leftData < rightData
TypeError: '<' not supported between instances of 'int' and 'str'
```
I am using Spyder 4.2.1 in MacOS Catalina. Also relevant:
```
Python 3.8.5 (default, Sep  4 2020, 02:22:02)
IPython 7.19.0 -- An enhanced Interactive Python.
```

**Comment by Tissot11** [comment]
Sorting by size does seem to work for me. But clicking on any variable in variable Explorer, the window resize automatically to smaller size.

**Comment by ccordoba12** [comment]
Thanks @AbhishekVermaDH for the reproducible example. I was able to reproduce this bug with it.

We'll try to fix it for our next version (4.2.2), to be released in February 15th.
